server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/step/ReportNewDuplicationMeasuresStepTest.java

```diff
@@ -0,0 +1,280 @@
+import static org.sonar.server.computation.task.projectanalysis.measure.Measure.newMeasureBuilder;
+import static org.sonar.server.computation.task.projectanalysis.measure.MeasureRepoEntry.entryOf;
+import static org.sonar.server.computation.task.projectanalysis.measure.MeasureRepoEntry.toEntries;
+import static org.sonar.server.computation.task.projectanalysis.measure.MeasureVariations.newMeasureVariationsBuilder;
+
+public class ReportNewDuplicationMeasuresStepTest {
+  private static final int ROOT_REF = 1;
+  private static final int MODULE_REF = 12;
+  private static final int SUB_MODULE_1_REF = 123;
+  private static final int SUB_MODULE_2_REF = 126;
+  private static final int DIRECTORY_REF = 1234;
+  private static final int FILE_1_REF = 12341;
+  private static final int FILE_2_REF = 12342;
+  private static final int FILE_3_REF = 1261;
+  private static final int FILE_4_REF = 1262;
+  private static final String SOME_FILE_KEY = "some file key";
+
+  @Rule
+  public TreeRootHolderRule treeRootHolder = new TreeRootHolderRule()
+    .setRoot(
+      builder(PROJECT, ROOT_REF)
+        .addChildren(
+          builder(MODULE, MODULE_REF)
+            .addChildren(
+              builder(MODULE, SUB_MODULE_1_REF)
+                .addChildren(
+                  builder(DIRECTORY, DIRECTORY_REF)
+                    .addChildren(
+                      builder(FILE, FILE_1_REF).build(),
+                      builder(FILE, FILE_2_REF).build())
+                    .build())
+                .build(),
+              builder(MODULE, SUB_MODULE_2_REF)
+                .addChildren(
+                  builder(FILE, FILE_3_REF).build(),
+                  builder(FILE, FILE_4_REF).build())
+                .build())
+            .build())
+        .build());
+  @Rule
+  public PeriodsHolderRule periodsHolder = new PeriodsHolderRule();
+  @Rule
+  public ScmInfoRepositoryRule scmInfoRepository = new ScmInfoRepositoryRule();
+  @Rule
+  public MetricRepositoryRule metricRepository = new MetricRepositoryRule()
+    .add(NEW_LINES_DUPLICATED);
+  @Rule
+  public MeasureRepositoryRule measureRepository = MeasureRepositoryRule.create(treeRootHolder, metricRepository);
+  @Rule
+  public DuplicationRepositoryRule duplicationRepository = DuplicationRepositoryRule.create(treeRootHolder);
+
+  @Before
+  public void setUp() {
+    periodsHolder.setPeriods(
+      new Period(2, "mode_p_1", null, parseDate("2009-12-25").getTime(), "u1"),
+      new Period(5, "mode_p_5", null, parseDate("2011-02-18").getTime(), "u2"));
+  }
+
+  NewDuplicationMeasuresStep underTest = new NewDuplicationMeasuresStep(treeRootHolder, periodsHolder, metricRepository, measureRepository, scmInfoRepository,
+    duplicationRepository);
+
+  @Test
+  public void compute_duplicated_lines_counts_lines_from_original_and_InnerDuplicate_of_a_single_line() {
+    TextBlock original = new TextBlock(1, 1);
+    duplicationRepository.addDuplication(FILE_1_REF, original, new TextBlock(2, 2));
```

Comment:
original should be replaced by new TextBlock(1, 1). No need to declare a variable for that.
(and do the same on following tests)